include/multi_blas_helper.cuh

```diff
@@ -320,12 +324,105 @@ namespace quda
       Spinor<x_store_t, Nx> W[NYW];
     };
 
-    template <typename T> struct coeff_array {
-      using type = T;
-      const T *data;
-      coeff_array() : data(nullptr) {}
-      coeff_array(const T *data) : data(data) {}
-    };
+    // the below are helpers used in the recursion
+
+    /**
+       @brief Split the set into two.  The optional parameter split is
+       used to specify the split point (needed for 2-d splitting).
+       @param[in] x The input set to split
+       @param[in] split The split point
+       @return Resulting split set
+    */
+    template <typename T> inline auto bisect(T &x, size_t split = 0)
```

Comment:
Harking back to some of my other comments, these functions could be put in the same separate header as the functions that create `set`s.

Reply:
I see the superficial similarity but I think it's best to keep the multi-blas set operations separate, at least for now.  These are dealing specifically with (implicit 2-d) sets of reference wrapped objects, whereas the above are dealing with the interface between lists of objects and reference wrappers.